apiserver/params/network.go

```diff
@@ -185,6 +189,13 @@ type NetworkConfig struct {
 
 	// IsDefaultGateway marks an interface that is a default gateway for a machine.
 	IsDefaultGateway bool `json:"is-default-gateway,omitempty"`
+
+	// NetworkOrigin represents the authoritative source of the NetworkConfig.
+	// It is expected that either the provider gave us this info or the
+	// machine gave us this info.
+	// Giving us this information allows us to reason about when a InterfaceInfo
+	// is in use.
+	NetworkOrigin NetworkOrigin `json:"origin,omitempty"`
```

Comment:
If we make a `NetworkConfigV2` we'll be in a special kind of hell to get this to work correctly for the version facade bump.